tests/functional/builtins/codegen/test_create_functions.py

```diff
@@ -105,7 +105,10 @@ def test2(a: uint256) -> Bytes[100]:
     with tx_failed():
         c.test2(0, gas=GAS_SENT)
 
+    # REVIEW: implied by tx_failed
     assert env.last_result["is_success"] is False
+    # REVIEW: add a comment maybe like "check the staticcall performed
+    # no sstore" (or reason string)
```

Comment:
i don't think it is, but that's one of the only reasons why a `staticcall` would consume all gas. i'm not sure why that `gasUsed` check is there tbh.

Reply:
I did not write this test, I don't know what were the initial reasons

Reply:
i guess i'm saying that as part of maintenance, we should do one of:
- explain why the line is there
- remove the line if we have determined it is useless
- explain that we don't understand why the line is there

this will make future maintenance easier because at least we have some documentation regarding what the line is doing

Reply:
This looks like it's testing the evm rather than the compiler :thinking:

Reply:
yea, it's a weird line